app/models/host/managed.rb

```diff
@@ -544,6 +544,30 @@ def apply_inherited_attributes(attributes, initialized = true)
     attributes
   end
 
+  def apply_interfaces_compute_attributes(attributes)
+    return attributes unless compute_resource
+    return attributes unless attributes[:interfaces_attributes]
+
+    attributes = hash_clone(attributes).with_indifferent_access
+    networks = compute_resource.networks
```

Comment:
I don't like that we match networks in Host::Managed as this might be compute_resource specific. I have an implementation ready where this is not required: https://github.com/timogoebel/foreman/commit/f672034e42f4c2246aa8d20aaed4e4a128e1eaae

Do you mind taking a look and updating this PR accordingly? Thanks.

Reply:
There is a small, but important, issue with this requested change.

The `compute_attributes` are used to determine whether a compute resource needs updating or not. See: `compute_update_required?` https://github.com/theforeman/foreman/blob/8397b3c93489ecf31d2b8cf88c50439709e29d50/app/models/concerns/orchestration/compute.rb#L232

Where `update_requried?` on the compares the key-values for changes. (https://github.com/theforeman/foreman/blob/develop/app/models/compute_resource.rb#L228) With this change it will always trigger a compute resource update since the network value will be something like 123456 (comes from the form) and not the actual name of the network.

An idea would be to change the form's select field to use the network's name as the value instead of the network id

Reply:
I changed the format of the compute attributes to always show the id and not the name. So that form and attributes are the same if nothing changes.